packages/wonder-stuff-sentry/package.json

```diff
@@ -4,12 +4,12 @@
     },
     "name": "@khanacademy/wonder-stuff-sentry",
     "version": "0.1.1",
-    "description": "",
+    "description": "Sentry integration support",
     "module": "dist/es/index.js",
     "main": "dist/index.js",
     "source": "src/index.js",
     "scripts": {
-        "test": "bash -c 'yarn --silent --cwd \"../..\" test $PWD'"
+        "test": "bash -c 'yarn --silent --cwd \"../..\" test ${@:0} $($([[ ${@: -1} = -* ]] || [[ ${@: -1} = bash ]]) && echo $PWD)'"
```

Comment:
woah, this is complicated, what is it doing?

Reply:
It is running the test script from the root `package.json`, but also allowing folks to specify test files to run by looking at the last arg and determining if it is not an option (i.e. doesn't start with `-`) and is not the `bash` command.

Really wish we could add comments to `package.json` files. :(

Reply:
To break it down, when in the `packages/wonder-stuff-sentry` folder and you run `yarn test`.

`@:0` = All the args that were after the `yarn test`. e.g. for `yarn test --coverage collate-sentry-data` this would be `--coverage collate-sentry-data`

`$([[ ${@: -1} = -* ]] || [[ ${@: -1} = bash ]])` = Does the last arg start with `-` or is it `bash` - this will give 0 or 1. This condenses down in the `$($([[ ${@: -1} = -* ]] || [[ ${@: -1} = bash ]]) && echo $PWD)` above to be either `$(0 && echo $PWD)` or `$(1 && echo $PWD)`. So, when the last arg starts with `-` or is `bash` we use the current working directory as our final arg, which runs all tests in our current package by generating a `yarn test` call against the root directory, passing the path of the package directory to test. If the last arg is not an option with `-` and is not `bash`, we don't add the package directory, deferring to whatever the entered last arg is, which allows things like `yarn test collate-sentry-data` to target specific tests.